clients/ui/frontend/src/app/App.tsx

```diff
@@ -1,20 +1,79 @@
-import * as React from 'react';
-import '@patternfly/react-core/dist/styles/base.css';
-import AppRoutes from '@app/AppRoutes';
-import '@app/app.css';
+import * as React from "react";
+import "@patternfly/react-core/dist/styles/base.css";
+import AppRoutes from "@app/AppRoutes";
+import "@app/app.css";
 import {
+  Alert,
+  Bullseye,
+  Button,
   Flex,
   Masthead,
   MastheadContent,
   MastheadToggle,
   Page,
+  PageSection,
   PageToggleButton,
-  Title
-} from '@patternfly/react-core';
-import NavSidebar from './NavSidebar';
-import { BarsIcon } from '@patternfly/react-icons';
+  Spinner,
+  Stack,
+  StackItem,
+  Title,
+} from "@patternfly/react-core";
+import NavSidebar from "./NavSidebar";
+import { BarsIcon } from "@patternfly/react-icons";
+import { AppContext } from './AppContext';
+import { PlatformType } from "../types";
 
 const App: React.FC = () => {
+  const username = "admin"; // TODO: replace with actual username
+  const userError = null; // TODO: replace with actual error
+  const config = {
+    username,
+  }; // TODO: replace with actual value
+  const configLoaded = true; // TODO: replace with actual value
+  const configError = null; // TODO: replace with actual value
+
+  // TODO: replace with actual value
+  const contextValue = {
+    config,
+    platform: PlatformType.KUBEFLOW,
+  };
+
+  // We lack the critical data to startup the app
+  if (userError || configError) {
+    // There was an error fetching critical data
+    return (
+      <Page>
+        <PageSection>
+          <Stack hasGutter>
+            <StackItem>
+              <Alert variant="danger" isInline title="General loading error">
+                <p>
+                  {(userError ? "User not loaded" : "Config not loaded") ||
+                    "Unknown error occurred during startup."}{" "}
+                  // TODO: replace with actual error message
```

Comment:
Not sure if you meant for this to render, but it's actually showing up in the UI. Based off the comment syntax I imagine you want this to be in JSX

Reply:
Yeah, that was just to get an error in the future, but I've refactored it, better to implement the customHook with real values and just refactor that.